sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java

```diff
@@ -448,6 +448,17 @@ public static class SubStatusCodes {
         public static final int NO_VALID_STORE_RESPONSE = 21009;
         public static final int SERVER_GENERATED_408 = 21010;
 
+        // SubStatusCodes for Client generated 500
+        public static final int MISSING_PARTITION_KEY_RANGE_ID_IN_CONTEXT = 21011;
```

Comment:
Please also register these substatuscodes - and validate uniqueness in msdata repository. We decided to keep msdata as "ultimate truth" for client-side substatus codes

Please also add them to statusCodes.md in java github (shared with custoemrs and css)

Reply:
|500|               0 |Unknown| `Internal Server error` - Error returned from server, Indicates unexpected and unqualified internal service error.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|500|   20902 - 20910 |Unknown| `Internal Server error` - Client generated 500. The error message will have the details about the cause.